cmd/tenant/main.go

```diff
@@ -57,3 +92,221 @@ func serve(c *cli.Context) (err error) {
+			return cli.Exit(err, 1)
+		}
+
+		var object ulid.ULID
+		if object, err = key.ObjectID(); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		if _, ok := orgKeys[object]; !ok {
+			missingTenants[key] = struct{}{}
+		}
+
+		return nil
+	}
+
+	if _, err = db.List(ctx, nil, nil, db.TenantNamespace, fetchTenants, nil); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Fetch all projects that do not have an organization key
+	missingProjects := make(map[db.Key]struct{})
+	fetchProjects := func(item *trtl.KVPair) error {
+		key := &db.Key{}
+		if err = key.UnmarshalValue(item.Key); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		var object ulid.ULID
+		if object, err = key.ObjectID(); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		if _, ok := orgKeys[object]; !ok {
+			// Projects are stored by tenantID:projectID, however we need the orgID for the index
+			project := &db.Project{}
+			if err = project.UnmarshalValue(item.Value); err != nil {
+				return cli.Exit(err, 1)
+			}
+
+			var orgKey db.Key
+			if orgKey, err = db.CreateKey(project.OrgID, project.ID); err != nil {
+				return cli.Exit(err, 1)
+			}
+
+			missingProjects[orgKey] = struct{}{}
+		}
+
+		return nil
+	}
+
+	if _, err = db.List(ctx, nil, nil, db.ProjectNamespace, fetchProjects, nil); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Create organization keys for all missing tenants and projects
+	var migratedTenants, migratedProjects int
+	var errs *multierror.Error
+	for tenant := range missingTenants {
+		var id ulid.ULID
+		if id, err = tenant.ObjectID(); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		var orgID ulid.ULID
+		if orgID, err = tenant.ParentID(); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		if dry {
+			fmt.Println("Missing org key for tenant", id.String())
+			continue
+		}
+
+		if err = db.PutOrgIndex(ctx, id, orgID); err != nil {
+			errs = multierror.Append(errs, err)
+			continue
+		}
+
+		migratedTenants++
+	}
+
+	for project := range missingProjects {
+		var id ulid.ULID
+		if id, err = project.ObjectID(); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		var orgID ulid.ULID
+		if orgID, err = project.ParentID(); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		if dry {
+			fmt.Println("Missing org key for project", id.String())
+			continue
+		}
+
+		if err = db.PutOrgIndex(ctx, id, orgID); err != nil {
+			errs = multierror.Append(errs, err)
+			continue
+		}
+
+		migratedProjects++
+	}
+
+	if errs != nil {
+		fmt.Println(errs.Error())
+	}
+
+	if dry {
+		fmt.Printf("%d tenants and %d projects would be migrated\n", len(missingTenants), len(missingProjects))
+	} else {
+		fmt.Println("Migrated", migratedTenants, "tenants and", migratedProjects, "projects")
+	}
+
+	return nil
+}
+
+func listKeys(c *cli.Context) (err error) {
+	var conf config.Config
+	if conf, err = config.New(); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	if err = db.Connect(conf.Database); err != nil {
+		return cli.Exit(err, 1)
+	}
```

Comment:
I know this is a quick script; but if we're going to have these commands be general purpose utilities, we should have `Before` and `After` -- this doesn't close the connection to the DB either. I'll make those changes and push them up so we can move on to the migration.